File: src/main/java/com/vaadin/demo/component/textfield/TextFieldStyles.java
package com.vaadin.demo.component.textfield;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.textfield.TextFieldVariant;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("text-field-styles")
public class TextFieldStyles extends HorizontalLayout {

    public TextFieldStyles() {
        setPadding(false);

        // tag::snippet[]
        TextField field = new TextField();
        field.addThemeVariants(TextFieldVariant.SMALL,
                TextFieldVariant.ALIGN_RIGHT,
                TextFieldVariant.HELPER_ABOVE);
        // end::snippet[]
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setValue("Value");

        add(field);
    }

    public static class Exporter extends DemoExporter<TextFieldStyles> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/textfield/text-field-styles.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/text-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('text-field-styles')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-text-field
        theme="align-right small helper-above-field"
        label="Label"
        helper-text="Helper text"
        value="Value"
        style="--vaadin-input-field-border-width: 1px;"
      ></vaadin-text-field>
      <!-- end::snippet[] -->
    `;
  }
}
